.github/workflows/automation-scripts/update-changelogs.js

```diff
@@ -0,0 +1,91 @@
+const fs = require('fs');
+const path = require('path');
+
+// Read all files in the directory
+function getFilePaths(directoryPath) {
+  let filePaths = [];
+  let files = fs.readdirSync(directoryPath);
+  files.forEach(file => {
+    let filePath = path.join(directoryPath, file);
+    filePaths.push(filePath);
+  });
+  return filePaths
+}
+
+function loadJsonFiles(filePath) {
+  // Load each JSON file
+  data = fs.readFileSync(filePath);
+  const jsonData = JSON.parse(data);
+  return jsonData;
+}
+
+// returns true if currentVersion is greater than incoming
+function compareVersions(currentVersion, incomingVersion) {
+  currentVersion = currentVersion.split('.').map(Number);
+  incomingVersion = incomingVersion.split('.').map(Number);
+  const currentDif = currentVersion.map((num, index) => {
+    return num - incomingVersion[index];
+  })
+  const leadNum = currentDif.filter((num) => { return num !== 0 })[0];
+  return leadNum > 0;
+}
+
+function findNewEntries(currentEntries, incomingEntries) {
+  currVersions = [];
+  incomingVersions = [];
+
+  currentEntries.forEach((entry) => {
+    currVersions.push(entry.version);
+  });
+
+  incomingEntries.forEach((entry) => {
+    incomingVersions.push(entry.version);
+  });
+
+  newEntries = [];
+  incomingVersions.forEach((version, i) => {
+    if (!currVersions.includes(version))
+      newEntries.push(incomingEntries[i]);
+  })
+
+  return newEntries;
+}
+
+function addEntriesToCurr(currentEntries, newEntries) {
+  let i = 0;
+  let j = 0;
+  while (j !== newEntries.length) {
+    if (!compareVersions(currentEntries[i].version, newEntries[j].version)) {
+      currentEntries.splice(i, 0, newEntries[j])
+      j++;
+    }
+    i++;
+  }
+  return currentEntries;
+}
+
+function fixChangeLogs(currentFiles, incomingFiles) {
+  numFiles = currentFiles.length;
+  for (i = 0; i < numFiles; i++) {
+    let currentJson = loadJsonFiles(currentFiles[i]);
```

Comment:
This can be `const` - const object properties can be reassigned, just not the object itself.

Reply:
gotcha thanks for the clarification

Reply:
done